mypy/subtypes.py

```diff
@@ -146,10 +146,9 @@ def visit_instance(self, left: Instance) -> bool:
             rname = right.type.fullname()
             if not left.type.has_base(rname) and rname != 'builtins.object':
                 return False
-
             # Map left type to corresponding right instances.
             t = map_instance_to_supertype(left, right.type)
-
+            # TODO: assert len(t.args) == len(right.args)
```

Comment:
The problem with `Awaitable` is fixed, the problem with `NewType` I'll wait for you to fix it in your PR as we discussed.